eng/common/scripts/Package-Properties.ps1

```diff
@@ -79,6 +82,61 @@ class PackageProps
         $this.Initialize($name, $version, $directoryPath, $serviceDirectory)
         $this.Group = $group
     }
+
+    hidden [object]GetValueSafely($hashtable, [string[]]$keys) {
+        $current = $hashtable
+        foreach ($key in $keys) {
+            if ($current.ContainsKey($key) -or $current[$key]) {
+                $current = $current[$key]
+            }
+            else {
+                return $null
+            }
+        }
+
+        return $current
+    }
+
+    hidden [HashTable]ParseYmlForArtifact([string]$ymlPath) {
+        if (Test-Path -Path $ymlPath) {
+            try {
+                $content = Get-Content -Raw -Path $ymlPath | CompatibleConvertFrom-Yaml
+                if ($content) {
+                    $artifacts = $this.GetValueSafely($content, @("extends", "parameters", "Artifacts"))
+
+                    $artifactForCurrentPackage = $artifacts | Where-Object { $_["name"] -eq $this.ArtifactName -or $_["name"] -eq $this.Name }
+
+                    if ($artifactForCurrentPackage) {
+                        return [HashTable]$artifactForCurrentPackage
```

Comment:
Multiple returns from the pipeline could trip this up. But the engsys would probably also be in a pretty messed up state if someone had multiple artifacts with the same name to begin with...

Reply:
Yeah if this happens it's going to hard error I believe. Because of the class usage the types actually matter, and if we try to convert an array to a HashTable I don't think that's going to work gracefully.

Let me try this with a deliberately broken artifact and let you know what happens.

Reply:
My hunch of it being a hard error due to type conversion is correct!

![image](https://github.com/user-attachments/assets/64a700d0-1fb0-415c-b0a0-ab7e85772f51)

```
Exception while parsing yml file C:\repo\azure-sdk-for-js\sdk\core\ci.yml: Cannot convert the "System.Object[]" value of type "System.Object[]" to type "System.Collections.Hashtable".
```

![image](https://github.com/user-attachments/assets/a1e8bfd8-9ba5-43c2-a307-330947c9aa01)